components/style/values/specified/image.rs

```diff
@@ -195,18 +195,31 @@ impl Gradient {
             _ => { return Err(()); }
         };
 
-        // https://drafts.csswg.org/css-images/#typedef-color-stop-list
-        if stops.len() < 2 {
-            return Err(())
-        }
-
         Ok(Gradient {
-            stops: stops,
+            items: items,
             repeating: repeating,
             gradient_kind: gradient_kind,
             compat_mode: compat_mode,
         })
     }
+
+    fn parse_items(context: &ParserContext, input: &mut Parser) -> Result<Vec<GradientItem>, ()> {
+        let mut seen_stop = false;
+        let items = try!(input.parse_comma_separated(|input| {
+            if seen_stop {
+                if let Ok(hint) = input.try(|i| LengthOrPercentage::parse(context, i)) {
+                    seen_stop = false;
+                    return Ok(GradientItem::InterpolationHint(hint));
+                }
+            }
+            seen_stop = true;
+            ColorStop::parse(context, input).map(GradientItem::ColorStop)
+        }));
+        if !seen_stop || items.len() < 2 {
```

Comment:
err, never mind, the `!seen_stop` check prevents it from ending in a interpolation hint.